packages/medusa/src/services/totals.js

```diff
@@ -34,6 +53,135 @@ class TotalsService extends BaseService {
     return subtotal
   }
 
+  /**
+   * Calculates shipping total
+   * @param {Cart | Object} object - cart or order to calculate subtotal for
+   * @return {int} shipping total
+   */
+  getShippingTotal(order) {
+    const { shipping_methods } = order
+    return shipping_methods.reduce((acc, next) => {
+      return acc + next.price
+    }, 0)
+  }
+
+  /**
+   * Calculates tax total
+   * Currently based on the Danish tax system
+   * @param {Cart | Object} object - cart or order to calculate subtotal for
+   * @return {int} tax total
+   */
+  async getTaxTotal(object) {
+    const subtotal = this.getSubtotal(object)
+    const shippingTotal = this.getShippingTotal(object)
+    const region = await this.regionService_.retrieve(object.region)
+    const { tax_rate } = region
+    return (subtotal + shippingTotal) * tax_rate
+  }
+
+  /**
+   * Calculates refund total of line items.
+   * If any of the items to return have been discounted, we need to
+   * apply the discount again before refunding them.
+   * @param {Order} order - cart or order to calculate subtotal for
+   * @param {[LineItem]} lineItems -
+   * @return {int} the calculated subtotal
+   */
+  async getRefundTotal(order, lineItems) {
+    const discount = order.discounts.find(
+      ({ discount_rule }) => discount_rule.type !== "free_shipping"
+    )
+
+    if (_.differenceBy(lineItems, order.items, "_id").length !== 0) {
+      throw new MedusaError(
+        MedusaError.Types.INVALID_ARGUMENT,
+        "Line items does not exist on order"
+      )
+    }
+
+    const subtotal = this.getSubtotal({ items: lineItems })
+
+    // if nothing is discounted, return the subtotal of line items
+    if (!discount) {
+      return subtotal
```

Comment:
I think we need to return the tax as well - for now we can just do `subtotal * region.tax_rate`.

Reply:
We could use `getTaxTotal` but that includes shipping also. And we prob. don't want that or do we ?

Reply:
Refunding shipping should come down to the store owner. For example, I could imagine a case where an owner would refund shipping if the item was defect or broken, while not refunding shipping if the customer was simply unhappy with the product. Let me know, what you think